### ISSUE TITLE
vault_aws_auth_backend_role no t appying policies when Vault provider version >= 2.2

### ISSUE THREAD

**Issue by josepplaltg** [terraform-providers/terraform-provider-vault#545]
Hi there,

Thank you for opening an issue. Please note that we try to keep the Terraform issue tracker reserved for bug reports and feature requests. For general usage questions, please see: https://www.terraform.io/community.html.

### Terraform Version
v0.11.14

### Affected Resource(s)
- vault_aws_auth_backend_role

If this issue appears to affect multiple resources, it may be an issue with Terraform's core, so please mention this.

### Terraform Configuration Files

# Vault role
resource "vault_aws_auth_backend_role" "kafkaconnect" {
  backend                  = "aws/"
  role                     = "${module.kafkaconnect.iam_role_id}"
  auth_type                = "iam"
  bound_iam_principal_arns = ["${module.kafkaconnect.iam_role_arn}"]
  ttl                = 900
  max_ttl            = 900
  policies           = ["default", "${lookup(var.connect_tags, "app")}-connect"]
}

### Expected Behavior
What should have happened?
The role should be created with 2 attached policies
### Actual Behavior
What actually happened?
The role is created without any policy attached
### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`

### Important Factoids
EC2 Classic

**Comment by tkinz27** [comment]
I believe I may be running into this issue to. I went to fix a deprecation notice, switching `policies` to `token_policies`, yet every time I try to do a `terraform apply` terraform keeps telling me it needs to reapply the change. I also see terraform trying to add `token_type` on every apply.

**Comment by ewold** [comment]
I can confirm to have to same problem. No policies are set, another apply wants to add them again, but nothing happens after the successful apply. Applying again still wants to add the policies to the role. 

```
Terraform v0.12.9
+ provider.aws v2.31.0
+ provider.consul v2.5.0
+ provider.local v1.4.0
+ provider.template v2.1.2
+ provider.tls v2.1.1
+ provider.vault v2.3.0
```

policies after the creation 
```
policies                          []
```

**Comment by moosilauke18** [comment]
Unfortunately your vault server version is probably lower than `1.2.0` where `token_*` was used: https://github.com/hashicorp/vault/blob/master/CHANGELOG.md#120-july-30th-2019

**Comment by josepplaltg** [comment]
@moosilauke18 True, I've 1.1.2, I'll fix the vault provider version to the 2.1 that is the last one that works for me until we upgrade the Vault server, thanks.
